annotation/tests/test_distribution.py

```diff
@@ -1277,3 +1291,238 @@ def test_get_page_number_combinations(
+        "users_default_load",
+        "users_overall_load",
+        "expected_output",
+        "expected_share_loads",
+    ),
+    (
+        (
+            [{"user_id": "0", "overall_load": 10, "default_load": 10}],
+            10,  # all_job_pages_sum
+            10.0,  # average_job_pages
+            10,  # users_default_load
+            10,  # users_overall_load
+            1.0,  # expected_output
+            [1],  # expected_share_loads
+        ),
+    ),
+)
+def test_find_users_share_loads(
+    users: List[DistributionUser],
+    all_job_pages_sum: int,
+    average_job_pages: float,
+    users_default_load: int,
+    users_overall_load: int,
+    expected_output: float,
+    expected_share_loads: List[float],
+):
+    output = find_users_share_loads(
+        users,
+        all_job_pages_sum,
+        average_job_pages,
+        users_default_load,
+        users_overall_load,
+    )
+    assert output == expected_output
+    for user, expected_share_load in zip(users, expected_share_loads):
+        assert user["share_load"] == expected_share_load
+
+
+@pytest.mark.parametrize(
+    ("files", "user_pages", "expected_output"),
+    (
+        (
+            [{"pages_number": 1, "file_id": 0}],
+            1,
+            [{"pages_number": 1, "file_id": 0}],
+        ),
+        (
+            [
+                {"pages_number": 10, "file_id": 0},
+                {"pages_number": 20, "file_id": 1},
+                {"pages_number": 30, "file_id": 2},
+                {"pages_number": 40, "file_id": 3},
+            ],
+            90,
+            [
+                {"pages_number": 20, "file_id": 1},
+                {"pages_number": 30, "file_id": 2},
+                {"pages_number": 40, "file_id": 3},
+            ],
+        ),
+    ),
+)
+def test_find_equal_files(
+    files: List[Dict[str, int]],
+    user_pages: int,
+    expected_output: List[Dict[str, int]],
+):
+    assert find_equal_files(files, user_pages) == expected_output
+
+
+@pytest.mark.parametrize(
+    ("files", "user_pages", "SPLIT_MULTIPAGE_DOC_setup", "expected_output"),
+    (
+        (
+            [{"pages_number": 10, "file_id": 0}],
+            20,
+            "",
+            ([{"pages_number": 10, "file_id": 0}], 0),
+        ),
+        (
+            [
+                {"pages_number": 10, "file_id": 0},
+                {"pages_number": 30, "file_id": 1},
+                {"pages_number": 0, "file_id": 2},
+            ],
+            20,
+            "a",
+            ([{"pages_number": 10, "file_id": 0}], 0),
+        ),
+    ),
+)
+def test_find_small_files(
+    files: List[Dict[str, int]],
+    user_pages: int,
+    SPLIT_MULTIPAGE_DOC_setup: str,
+    expected_output,
+):
+    def find_files_for_task_side_effect(files, pages_for_task):
+        lst = []
+        for file in files:
+            if file["pages_number"] and file["pages_number"] in pages_for_task:
+                lst.append(file)
+        return lst
```

Comment:
too complicated. if you do that for your patch, it is most likely due to incorrect patching. think what you could patch to avoid such complex thing

Reply:
Now that I think about it, we don't need to patch the function _find_files_for_task_. I will remove this patch and it will make the test much simpler

Reply:
Also, I've just now realized that my test doesn't cover one edge case. I will add another test case for this test